import-core/import-services/src/main/java/org/devgateway/importtool/services/processor/IATIProcessor.java

```diff
@@ -405,4 +643,407 @@ private String getPropertieFieldName(String lang,String langPackType) {
+                if (includeCountry) {
+                    recipient.setCode(code);
+                    Optional<FieldValue> fieldValue = field.getPossibleValues().stream().filter(f -> f.getCode().equals(code)).findFirst();
+                    if (fieldValue.isPresent()) {
+                        recipient.setValue(fieldValue.get().getValue());
+                    }
+                    recipient.setPercentage(fieldElement.getAttribute("percentage"));
+                    recipients.add(recipient);
+                }
+            }
+            document.addRecepientCountryFields(field.getFieldName(), recipients);
+        };
+        processForEachFilteredByType(recipientCountry, FieldType.RECIPIENT_COUNTRY);
+    }
+
+    protected void processLocationElementType(InternalDocument document, Element element) {
+        Consumer<Field> locationConsumer = field -> {
+            NodeList fieldNodeList = element.getElementsByTagName(field.getFieldName());
+            List<String> codesLocation = new ArrayList<>();
+            for (int j = 0; j < fieldNodeList.getLength(); j++) {
+                Element fieldElement = (Element) fieldNodeList.item(j);
+                String name = getNameFromElement(fieldElement);
+
+                if (!StringUtils.isBlank(name)) {
+                    codesLocation.add(name);
+                    FieldValue fv = new FieldValue();
+                    if (name != null && !name.isEmpty()) {
+                        fv.setCode(name);
+                        fv.setValue(name);
+                        fv.setSelected(true);
+                    }
+                    int index = field.getPossibleValues() == null ? 0 : field.getPossibleValues().size();
+                    fv.setIndex(index);
+                    if (field.getPossibleValues() == null) {
+                        field.setPossibleValues(new ArrayList<FieldValue>());
+                    }
+                    if (!field.getPossibleValues().stream().anyMatch(n -> {
+                        return n.getCode().equals(fv.getValue());
+                    })) {
+                        field.getPossibleValues().add(fv);
+                    }
+                }
+                if (!codesLocation.isEmpty()) {
+                    String[] codeValues = codesLocation.stream().toArray(String[]::new);
+                    document.addStringMultiField(field.getFieldName(), codeValues);
+                }
+            }
+        };
+        processForEachFilteredByType(locationConsumer, FieldType.LOCATION);
+    }
+
+    protected void processDateElementType(InternalDocument document, Element element) throws ParseException {
+        NodeList nodes = element.getElementsByTagName("activity-date");
+        for (int j = 0; j < nodes.getLength(); ++j) {
+            Element e = (Element) nodes.item(j);
+            Field field = getField(e, FieldType.DATE);
+            String localDate = e.getAttribute("iso-date");
+                String format = "yyyy-MM-dd";
```

Comment:
Please change to use the constant ISO_DATE defined in the same class.

Reply:
Done